CRM/Core/BAO/CustomField.php

```diff
@@ -1214,6 +1214,9 @@ public static function getDisplayValueCommon(
       case 'AdvMulti-Select':
       case 'Multi-Select':
         if (is_array($value)) {
+          // CRM-12989 fix
+          CRM_Utils_Array::formatArrayKeys($value);
+
```

Comment:
Are you sure it's appropriate to do this for type 'Multi-Select'? I think those values should already be formatted properly.